src/taskgraph/util/vcs.py

```diff
@@ -172,15 +198,10 @@ def branch(self):
 
     @property
     def remote_name(self):
-        remotes = self.run("paths", "--quiet").splitlines()
-        if len(remotes) == 1:
-            return remotes[0]
-
-        if "default" in remotes:
-            return "default"
-
-        raise RuntimeError(
-            f"Cannot determine remote repository name. Candidate remotes: {remotes}"
+        return self._get_most_suitable_remote(
+            ("paths", "--quiet"),
```

Comment:
What about if instead of passing in a `remote_command`, each vcs has:
```
@property
def remotes():
```
?

Reply:
Similarly I wonder if `default_remote_name` should A) be a list of remote names, and B) be an attribute on the class.

Reply:
These are excellent points! I didn't like what I made with `remote_command`. I find your suggestion much more elegant! 💯 

Done